modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/JavaUpgradeModelPermissionsCheck.java

```diff
@@ -0,0 +1,263 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.check.util.SourceUtil;
+import com.liferay.source.formatter.parser.JavaTerm;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Michael Cavalcanti
+ */
+public class JavaUpgradeModelPermissionsCheck extends BaseJavaTermCheck {
+
+	@Override
+	protected String doProcess(
+			String fileName, String absolutePath, JavaTerm javaTerm,
+			String fileContent)
+		throws Exception {
+
+		List<String> importNames = javaTerm.getImportNames();
+
+		String javaTermContent = javaTerm.getContent();
+
+		if (!importNames.contains(
+				"com.liferay.portal.kernel.service.ServiceContext")) {
+
+			return javaTermContent;
+		}
+
+		boolean hasSetGroupPermissions = false;
+
+		Matcher setGroupPermissionsMatcher =
+			_setGroupPermissionsPattern.matcher(javaTermContent);
+
+		if (setGroupPermissionsMatcher.find()) {
+			hasSetGroupPermissions = true;
+		}
+
+		boolean hasSetGuestPermissions = false;
+
+		Matcher setGuestPermissionsMatcher =
+			_setGuestPermissionsPattern.matcher(javaTermContent);
+
+		if (setGuestPermissionsMatcher.find()) {
+			hasSetGuestPermissions = true;
+		}
+
+		if ((hasSetGroupPermissions || hasSetGuestPermissions) &&
+			javaTerm.isJavaClass()) {
+
+			if (!fileContent.contains(
+					".portal.kernel.service.permission.ModelPermissions")) {
```

Comment:
Use the fully qualified name: "com.portal.kernel.service.permission.ModelPermissions".

Reply:
Ok